src/uwtools/yaml_file.py

```diff
@@ -0,0 +1,50 @@
+# (C) Copyright 2020-2022 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Part of this software is developed by the Joint Center for Satellite Data Assimilation (JCSDA) together with its partners.
+#
+# (C) Copyright 2022-2022 NOAA
+import yaml
+from .nice_dict import NiceDict
+
+class YAMLFile(NiceDict):
+
+    """
+        Reads a YAML file as a NiceDict and recursively converts
+        nested dictionaries into NiceDict.
+        Provides a way of saving the dictionary issued from the yaml file, after modification or not.
+    """
+
+    def __init__(self, config_file=None, data=None):
+        super().__init__()
+        if config_file is not None:
+            with open(config_file) as f:
+                config = yaml.load(f, Loader=yaml.Loader)
+        else:
+            config = data
+        if config is not None:
+            self.update(self._configure(config))
+
+    def _configure(self, config):
+        for key, value in config.items():
+            if isinstance(value, dict):
+                config[key] = NiceDict(value)
+                self._configure(value)
+            elif isinstance(value, list):
+                for i, v in enumerate(value):
+                    if isinstance(v, dict):
+                        value[i] = NiceDict(v)
+                        self._configure(v)
+        return config
+
+    def save(self, target):
```

Comment:
No validation on `target` (string, path.Pathlib, etc).  No checks.  

Saving to a file is outside the scope of this class.  A UD (User/Developer) could call `dump` or `__str__` and write the file themselves, alleviating us of the responsibility.